backend/infrahub/core/schema_manager.py

```diff
@@ -523,6 +523,30 @@ def generate_identifiers(self) -> None:
                 rel.identifier = str("__".join(sorted([node.kind, rel.peer]))).lower()
             self.set(name=name, schema=node)
 
+    @staticmethod
+    def validate_attr_or_mandatory_rel_path(
+        schema_path: SchemaAttributePath, node_schema: MainSchemaTypes, element_name: str, element_path: str
```

Comment:
Have you considered using the function `SchemaBranch.validate_schema_path` here ? it has been designed specifically so you can provide the criteria using flags

Reply:
Yes, but some criteria seem to be significantly specific to certain fields. For instance, `uniqueness_constraints` raises on optional relationship while `order_by` should not, so it might be relevant to have specific logic in dedicated functions